checkbox-ng/plainbox/provider_manager.py

```diff
@@ -551,7 +551,13 @@ def register_parser(self, subparsers):
 
     def invoked(self, ns):
         pp_env = os.getenv("PROVIDERPATH")
-        if pp_env and not os.path.samefile(pp_env, ns.directory):
+        try:
+            samefile = pp_env is not None and os.path.samefile(
+                pp_env, ns.directory
+            )
+        except FileNotFoundError:
+            samefile = False
+        if pp_env and not samefile:
```

Comment:
The logic is a bit confusing here, I think it would be clearer like this
```suggestion
        # If $PROVIDERPATH is defined, use it instead of the one on the
        # namespace
        if pp_env:
            try:
                samefile = os.path.samefile(pp_env, ns.directory)
            except FileNotFoundError:
                samefile = False
            if not samefile:
                _logger.warning(
                    "$PROVIDERPATH is defined, ignoring -d/--directory"
                    " and developing in: %s", pp_env
                )
```